catalog/app/utils/AWS/S3.js

```diff
@@ -51,21 +47,27 @@ export const useRequest = (extra) => {
   const authenticated = reduxHook.useMappedState(Auth.selectors.authenticated)
   return React.useMemo(
     () => ({ bucket, operation, params }) => {
-      let client
-      if (!authenticated && operation === 'selectObjectContent') {
-        client = s3SelectClient
-      } else if (cfg.shouldProxy(bucket)) {
-        client = proxyingClient
-      } else {
-        client = regularClient
-      }
+      const client =
+        !authenticated && operation === 'selectObjectContent'
+          ? s3SelectClient
+          : regularClient
       const method =
-        authenticated && cfg.shouldSign(bucket)
+        cfg.mode === 'LOCAL' || (authenticated && cfg.shouldSign(bucket))
           ? 'makeRequest'
           : 'makeUnauthenticatedRequest'
-      return client[method](operation, params).promise()
+      const req = client[method](operation, params)
+      if (cfg.shouldProxy(bucket) && client === regularClient) {
+        req.on('sign', () => {
+          // *After* the request has been signed with the original S3 hostname / path,
+          // change it to the proxy. Proxy will then change it back while keeping the
+          // original signature.
+          req.httpRequest.endpoint = new AWS.Endpoint(cfg.s3Proxy)
+          req.httpRequest.path = `/${bucket}${req.httpRequest.path}`
```

Comment:
@nl0 @dimaryaz: I think we need to pass the region along with the bucket name to the proxy here so that the proxy can generate the right hostname for buckets in regions other than `us-east-1`.